packages/salesforcedx-vscode-core/src/commands/forceApexExecute.ts

```diff
@@ -126,41 +131,40 @@ export class AnonApexGatherer
   }
 }
 
-export class ApexLibraryExecuteExecutor extends ApexLibraryExecutor {
-  protected executeService: ExecuteService | undefined;
+export class ApexLibraryExecuteExecutor extends LibraryCommandletExecutor<ApexExecuteParameters> {
+  protected executionName = nls.localize('apex_execute_text');
+  protected logName = 'force_apex_execute_library';
 
-  public createService(conn: Connection): void {
-    this.executeService = new ExecuteService(conn);
-  }
+  private diagnostics = vscode.languages.createDiagnosticCollection(
+    'apex-errors'
+  );
 
-  public async execute(
-    response: ContinueResponse<{ fileName?: string; apexCode?: string }>
-  ): Promise<void> {
-    this.setStartTime();
+  protected async run(response: ContinueResponse<ApexExecuteParameters>): Promise<boolean> {
+    const connection = await workspaceContext.getConnection();
+    const executeService = new ExecuteService(connection);
+    const { apexCode, fileName: apexFilePath } = response.data;
 
-    try {
-      await this.build(
-        nls.localize('apex_execute_text'),
-        nls.localize('force_apex_execute_library')
-      );
+    const result = await executeService.executeAnonymous({ apexFilePath, apexCode });
 
-      if (this.executeService === undefined) {
-        throw new Error('ExecuteService is not established');
-      }
+    const success = result.compiled && result.success;
```

Comment:
do we need to check for both `compiled` and `success`? i think `success` is only true if there's no compile or runtime issues

Reply:
Ah nice, yeah probably not. This was the logic taken from ApexLibraryExecutor